src/core/etl/src/Flow/ETL/Row/Entry/JsonEntry.php

```diff
@@ -109,30 +94,39 @@ public function isEqual(Entry $entry) : bool
 
     public function map(callable $mapper) : Entry
     {
-        return self::fromJsonString($this->name, $mapper($this->value()));
+        return new self($this->name, $mapper($this->value()), $this->object);
     }
 
     public function name() : string
     {
         return $this->name;
     }
 
-    public function rename(string $name) : Entry
+    public function phpType() : Type
     {
-        $entry = new self($name, $this->value);
-        $entry->object = $this->object;
+        if ([] === $this->value) {
+            return ArrayType::empty();
+        }
 
-        return $entry;
+        return new ArrayType();
```

Comment:
I think this should be a StringType, because now on the one hand `phpType()` is array but `value()` return type is string, and `phpType()` should be aligned with return type of `value()`